components/channel_selector_modal/channel_selector_modal.tsx

```diff
@@ -1,48 +1,54 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
+import {ChannelSearchOpts, ChannelWithTeamData} from 'mattermost-redux/types/channels';
+
+import {ActionResult} from 'mattermost-redux/types/actions';
+
 import Constants from 'utils/constants';
 import {localizeMessage, compareChannels} from 'utils/utils.jsx';
 
-import MultiSelect from 'components/multiselect/multiselect';
+import MultiSelect, {Value} from 'components/multiselect/multiselect';
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 
-const CHANNELS_PER_PAGE = 50;
-
-export default class ChannelSelectorModal extends React.PureComponent {
-    static propTypes = {
-        searchTerm: PropTypes.string.isRequired,
-        onModalDismissed: PropTypes.func,
-        onChannelsSelected: PropTypes.func,
-        groupID: PropTypes.string.isRequired,
-        actions: PropTypes.shape({
-            loadChannels: PropTypes.func.isRequired,
-            setModalSearchTerm: PropTypes.func.isRequired,
-            searchAllChannels: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    searchTerm: string,
+    onModalDismissed?: () => void,
+    onChannelsSelected: (channels:ChannelWithTeamData[]) => void,
+    groupID: string,
+    actions: {
+        loadChannels: (page?: number, perPage?: number, notAssociatedToGroup?: string, excludeDefaultChannels?: boolean) => Promise<{data: ChannelWithTeamData[]}>,
+        setModalSearchTerm: (term: string) => ActionResult,
+        searchAllChannels: (term: string, opts?: ChannelSearchOpts) => Promise<{data: ChannelWithTeamData[]}>,
+    },
+}
 
-    constructor(props) {
-        super(props);
+type State = {
+    values: ChannelWithTeamData[],
```

Comment:
Should this be an array of `ChannelWithTeamData & Value`?

Reply:
I've updated the MR with this suggestion.